### ISSUE TITLE
Required Parameter name not stated in error message for @BodyParam and some others

### ISSUE THREAD

**Issue by IAMtheIAM** [pleerock/routing-controllers#280]
For POST method, the required parameter is not listed in the error message as you have shown for GET

```
    "message": "Parameter is required for request on POST /api/login/getBearerToken",
```

GET
![GET](https://user-images.githubusercontent.com/10618781/30133480-52974b40-9354-11e7-80a9-2a69bab3aa2a.png)

POST
![post](https://user-images.githubusercontent.com/13709948/30187707-a54b829e-93df-11e7-91f4-f1f0b921aedd.png)

It should display the required paramater that is missing like in the example for GET, so that the client knows what wen't wrong, in case they don't have access to the nodejs code

**Comment by NoNameProvided** [comment]
Hi @IAMtheIAM 

What decorators do you use? In your first `GET` example you get a query parameter error what works the same for `POST` requests as well.

<img width="1014" alt="screen shot 2017-09-08 at am 6 20 37" src="https://user-images.githubusercontent.com/4706265/30196023-e274d674-945d-11e7-94c2-95abe707e439.png">

In your second example the error comes most likely because a `@BodyParam` decorator, am I right? So you want see what body param is missing when using the  `@BodyParam` decorator?

**Comment by 19majkel94** [comment]
> In your second example the error comes most likely because a @BodyParam decorator, am I right? So you want see what body param is missing when using the @BodyParam decorator?

Indeed. We have introduced need param types in decorators:
```ts
export type ParamType = "body"
    |"body-param"
    |"query"
    |"queries"
    |"header"
    |"headers"
    |"file"
    |"files"
    |"param"
    |"params"
    |"session"
    |"state"
    |"cookie"
    |"cookies"
    |"request"
    |"response"
    |"context"
    |"current-user"
    |"custom-converter";
```
But we forget to update the param required error body:
```ts
switch (param.type) {
    case "param":
        paramName = `Parameter "${param.name}" is`;
        break;

    case "body":
        paramName = "Request body is";
        break;

    case "query":
        paramName = `Query parameter "${param.name}" is`;
        break;

    case "header":
        paramName = `Header "${param.name}" is`;
        break;

    case "file":
        paramName = `Uploaded file "${param.name}" is`;
        break;

    case "files":
        paramName = `Uploaded files "${param.name}" are`;
        break;

    case "session":
        paramName = "Session is";
        break;

    case "cookie":
        paramName = "Cookie is";
        break;

    default:
        paramName = "Parameter is";
}
```

**Comment by IAMtheIAM** [comment]
Yes, I am using `@BodyParam` decorator, so I would like to see which body param is missing in error method. Thanks :D

**Comment by 19majkel94** [comment]
@IAMtheIAM 
Go ahead and send a PR! I've showed you when to dig in the code 😉

**Comment by IAMtheIAM** [comment]
So perhaps we just need this 

```
switch (param.type) {
    case "param":
        paramName = `Parameter "${param.name}" is`;
        break;

    case "body":
        paramName = "Request body is";
        break;

    case "body-param":
        paramName = `Body parameter "${param.name}" is`;
        break;

    case "query":
        paramName = `Query parameter "${param.name}" is`;
        break;

    case "header":
        paramName = `Header "${param.name}" is`;
        break;

    case "file":
        paramName = `Uploaded file "${param.name}" is`;
        break;

    case "files":
        paramName = `Uploaded files "${param.name}" are`;
        break;

    case "session":
        paramName = "Session is";
        break;

    case "cookie":
        paramName = "Cookie is";
        break;

    default:
        paramName = "Parameter is";
}
```

**Comment by IAMtheIAM** [comment]
I will start PR :)

**Comment by IAMtheIAM** [comment]
[PR submitted](https://github.com/pleerock/routing-controllers/pull/284) - are there any CI tests?

**Comment by NoNameProvided** [comment]
Yes there are, they are passed, PR looked good so merged. Thanks for the contribution 🎉

**Comment by NoNameProvided** [comment]
Fixed in #284

**Comment by IAMtheIAM** [comment]
You're welcome, thanks too.
